### ISSUE TITLE
Unwanted ragment placeholder on scalar types in sample query

### ISSUE THREAD

**Issue by tuananhtd** [anvilco/spectaql#385]
This is the sample query generated for one of my queries.
```
query getClose($id: String!) {
  getClose(id: $id) {
    id {
      ...StringFragment
    }
    fund {
      ...FundSubscriptionFragment
    }
    name {
      ...StringFragment
    }
    targetDate {
      ...DateTimeFragment
    }
  }
}
```
The problem is `id`, `name` and `targetDate` fields are just scalar type (`string` and `DateTime`) so I wish it could be like this.
```
query getClose($id: String!) {
  getClose(id: $id) {
    id
    fund {
      ...FundSubscriptionFragment
    }
    name
    targetDate
  }
}
```

I tried to put `fieldExpansionDepth` to 2 or 3 but nothing changed. Any idea where did I mess things up?

**Comment by tuananhtd** [comment]
Thank @newhouse for the awesome tools! Please find the attached file here: [config.zip](https://github.com/anvilco/spectaql/files/8860800/config.zip)

**Comment by newhouse** [comment]
@tuananhtd the problem is that in your Scalar definitions in your `instrospection.json`,  for some reason the `fields` key is an empty array (`"fields": []`) when (in my experience) it should be `null`. SpectaQL is checking for `null` there to determine if it's a Scalar that should not be expanded, or if it's a complex object that should be `...SomeSortOfFragment`.

What is generating your `introspection.json` file? Are you doing some manipulation yourself? It's an easy, safe "fix" for me on the SpectaQL side, but if what you're doing is outside of the specifications, I am hesitant for SpectaQL to handle this and would instead ask that you fix things on your side.

Let me know!

**Comment by tuananhtd** [comment]
Thank you for helping to debug this. I'm using `Dgraph` which in turn uses the scene using `Apolo` behind the scene. I'll do some `JSON` manipulation myself. Not sure about `Apolo` implementation but it also puts a `directive` section in the introspection response and causes some trouble with `SpectaQL` cause we don't have a config to un-document it yet.

**Comment by newhouse** [comment]
I've decided to fix and publish this for you in version `1.2.3`. Enjoy!
